lib/core/services/sort-by-category.service.spec.ts

```diff
@@ -0,0 +1,133 @@
+ * limitations under the License.
+ */
+
+import { setupTestBed } from '../testing/setup-test-bed';
+import { CoreTestingModule } from '../testing/core.testing.module';
+import { TranslateModule } from '@ngx-translate/core';
+import { SortbleByCategoryItem, SortByCategoryMapperService } from './sort-by-category.service';
+
+interface TestSortbleByCategoryItem extends SortbleByCategoryItem {
+    id: string;
+}
+
+describe('SortByCategoryService', () => {
+
+    let mapper: SortByCategoryMapperService<TestSortbleByCategoryItem>;
+
+    const DEFAULT_CATEGORIES = [
+        '',
+        'http://bpmn.io/schema/bpmn',
+        'http://www.activiti.org/processdef',
+        'http://www.activiti.org/test'
+      ];
+
+    setupTestBed({
+        imports: [
+            TranslateModule.forRoot(),
+            CoreTestingModule
+        ]
+    });
+
+    beforeEach(() => {
+        mapper = new SortByCategoryMapperService();
+    });
+
+    it('should map items By categories ', () => {
+
+    const items: TestSortbleByCategoryItem[] = [{ 'id': 'id1',  'name': 'firstCategory_222', 'category': 'category1' },
+                                                { 'id': 'id2',  'name': 'secondCategory_AA', 'category': 'category2' },
+                                                { 'id': 'id3',  'name': 'firstCategory_111', 'category': 'category1' },
+                                                { 'id': 'id4',  'name': 'secondCategory_BB', 'category': 'category2' },
+                                                { 'id': 'id5',  'name': 'Default_a', 'category': 'http://bpmn.io/schema/bpmn' },
+                                                { 'id': 'id6',  'name': 'Default_b', 'category': '' }];
+ 
+        const expectedContentsByCategory = [{ category: 'category1', items: [items[2], items[0]] },
+                                            { category: 'category2', items: [items[1], items[3]] },
+                                            { category: '', items: [items[4], items[5]] }];
+
+        const result = mapper.mapItems(items, DEFAULT_CATEGORIES);
+
+        expect(result).toEqual(expectedContentsByCategory);
+    });
+
+    it('should display items without category if all of the items have default category', () => {
+
+        const defaulValues: TestSortbleByCategoryItem[] = [{
+            'name': 'name-b',
+            'id': 'id',
+            'category': 'http://bpmn.io/schema/bpmn'
+    
+        }, {
+            'name': 'name-b',
+            'id': 'id2',
+            'category': 'http://www.activiti.org/test'
+        }, { 
+            'name': 'name-c',
+            'id': 'id3',
+            'category': ''
+        }]
+
+        const result = mapper.mapItems(defaulValues, DEFAULT_CATEGORIES);
+
+        expect(result.length).toBe(1);
+        expect(result[0].category).toBe('');
+    });
+
+    it('should display items in ascending order under no category label if there is at least one other category provided', () => {
```

Comment:
As above, this service is mapping one collection to another, it is not displaying anything